packages/binding-coap/src/coap-client.ts

```diff
@@ -219,7 +219,10 @@ export default class CoapClient implements ProtocolClient {
             hostname: requestUri.hostname ?? "",
             port: requestUri.port != null ? parseInt(requestUri.port, 10) : 5683,
             pathname: requestUri.pathname ?? "",
-            query: requestUri.query ?? "",
+            query:
+                requestUri.search && requestUri.search.length > 0 && requestUri.search[0] === "?"
+                    ? requestUri.search.substring(1)
+                    : "",
```

Comment:
It is a bit clumsy since `requestUri.search` has a leading `?` while the _old_ `requestUri.query` did not have it

`?v=6xJ`
vs
`v=6xJ`

@JKRhb please have a look